src/updater/updater.cpp

```diff
@@ -583,8 +583,15 @@ Updater::LoadAndUpdateEdgeExpandedGraph(std::vector<extractor::EdgeBasedEdge> &e
             [&] { extractor::files::readNodeData(config.GetPath(".osrm.ebg_nodes"), node_data); },
 
             [&] {
-                extractor::files::readTurnWeightPenalty(
-                    config.GetPath(".osrm.turn_weight_penalties"), turn_weight_penalties);
+                // we are going to overwrite this `.turn_weight_penalties` afterwards,
```

Comment:
This code modifies a couple of other files too and probably we should use similar approach for them too, but it is not so obvious as here: IIUC other files are related to traffic handling, i.e. it is more obvious for users that they need to provide data for all edges while updating speeds(but tbh I just had brief look into that code and not confident in it :) ).

Reply:
On another hand it seems to be good idea to preserve "default" speed weights for segments user didn't provide data for 🤔 So would appreciate opinions here.

Reply:
@mjjbell wdyt?

Reply:
The current behaviour of overwriting the files on updates means that you can make multiple independent changes and they will all apply. 

This proposed change would alter that behaviour (only last update is applied), with the alternative being to apply the cumulative changes on every update, which will be slower. So I don't think we should go with this.

For someone who is updating turn weights via traffic updates, we can probably assume they have a model of the data for any time, so aren't too bothered about recovering the original weights.

For conditional penalties, I think what we need is an `else` case that can set the turn weight penalty back to the previous value.

https://github.com/Project-OSRM/osrm-backend/blob/2cf957148bc01a7dc31ba396b6c8c3c28117ae8d/src/updater/updater.cpp#L514-L521

Alternatively, rather then overwriting the weight value when the conditional is applied, we only set it in the local value used to update the edge weights? That way we aren't destroying the value used when the conditional doesn't apply.

https://github.com/Project-OSRM/osrm-backend/blob/2cf957148bc01a7dc31ba396b6c8c3c28117ae8d/src/updater/updater.cpp#L765